lxd/vm_qemu.go

```diff
@@ -2376,13 +2377,61 @@ func (vm *vmQemu) RenderState() (*api.InstanceState, error) {
 		status, err := vm.agentGetState()
 		if err != nil {
 			logger.Warn("Could not get VM state from agent", log.Ctx{"project": vm.Project(), "instance": vm.Name(), "err": err})
-		} else {
-			status.Pid = int64(pid)
-			status.Status = statusCode.String()
-			status.StatusCode = statusCode
+			status = &api.InstanceState{}
+			status.Processes = -1
+
+			networks := map[string]api.InstanceStateNetwork{}
+			for k, m := range vm.ExpandedDevices() {
+				// We only care about nics.
+				if m["type"] != "nic" || m["nictype"] != "bridged" {
+					continue
+				}
+
+				// Fill the MAC address.
+				m, err := vm.fillNetworkDevice(k, m)
+				if err != nil {
+					return nil, err
+				}
+
+				// Parse the lease file.
+				addresses, err := networkGetLeaseAddresses(vm.state, m["parent"], m["hwaddr"])
+				if err != nil {
+					return nil, err
+				}
+
+				if len(addresses) == 0 {
+					continue
+				}
 
-			return status, nil
+				// Get MTU.
+				iface, err := net.InterfaceByName(m["parent"])
+				if err != nil {
+					return nil, err
+				}
+
+				if m["host_name"] == "" {
+					m["host_name"] = vm.localConfig[fmt.Sprintf("volatile.%s.host_name", k)]
+				}
+
+				networks[k] = api.InstanceStateNetwork{
+					Addresses: addresses,
+					Counters:  api.InstanceStateNetworkCounters(shared.NetworkGetCounters(m["host_name"])),
```

Comment:
@stgraber so the network counters look like they are the wrong way around compared to the data sent from the agent.

Example without agent running:

```
  Network usage:
    eth0:
      Bytes received: 123.89kB
      Bytes sent: 3.94MB
      Packets received: 1718
      Packets sent: 2229
```

Example with agent running:

```
    enp5s0:
      Bytes received: 3.93MB
      Bytes sent: 124.14kB
      Packets received: 2205
      Packets sent: 1722
```

Aside from the interface names being different (this is a separate, perhaps unavoidable issue), the byte counters are swapped in direction.

When doing `lxc info`, should the byte counters being from the host's perspective or the instance's perspective?

CC @monstermunchkin

Reply:
Oh right, we need to swap them since I pull them from the host side.

Reply:
For containers we report usage from within the container, so we should be doing the same for VM, swapping the counters in this code will do that.